### ISSUE TITLE
Can't use Kubernetes delete cascade orphan

### ISSUE THREAD

**Issue by dacofr** [fiaas/fiaas-deploy-daemon#261]
Hi :wave: 

With the current implementation of Fiaas, we cannot use :
`kubectl delete fa bla-bla --cascade=orphan` 

Some objects like PDB, RoleBindings, SA are well orphaned after deletion (owner reference is removed and objects still remain clusters), but that's not the case for Deployment, Service, Ingress because there is an explicit deletion that is done here : 

```
    def delete(self, app_spec: AppSpec):
        self._ingress_deployer.delete(app_spec)
        self._autoscaler_deployer.delete(app_spec)
        self._service_deployer.delete(app_spec)
        self._deployment_deployer.delete(app_spec)
```
https://github.com/fiaas/fiaas-deploy-daemon/blob/master/fiaas_deploy_daemon/deployer/kubernetes/adapter.py#L69

This made sense in the past, but now that each objects has an owner reference, we can let Kubernetes do the garbage collection for us.

I will send a PR to remove this explicit deletion, WDYT ?

Thanks

**Comment by perj** [comment]
Hi @dacofr!

I think that makes a lot of sense to support all three deletion modes.

If I'm reading [the docs](https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/#ownership-and-finalizers) (also [this one](https://kubernetes.io/docs/tasks/administer-cluster/use-cascading-deletion/)) correctly the controller (fiaas-deploy-daemon) should still delete the children in foreground mode.
Thus the code should check the finalizers of the application and still run the above code if it's set to foreground, but not run if it's unset. In orphan mode, it still needs to update the children to remove the ownerReference, I belive, I don't see anything in the docs about that being automatic.

When using foreground of orphan mode, it seems the controller is [expected](https://kubernetes.io/docs/concepts/overview/working-with-objects/finalizers/) to update the Application object and remove those finalizers. This part also need to be implemented for it to work fully. But I think if the children still has the ownerReference when that happens, they'll still be automatically deleted. Do test that part though.

**Comment by dacofr** [comment]
Hello

@perj and @oyvindio, thanks both for your feedback :+1: 
Yes during my tests, simply commenting out the four lines of explicit deletion was enough to have all three modes of deletion to work as expected. 

Didn't find this existing PR. 
But @oyvindio, If you want, I can start a new fresh PR and do the same kind of stuff.

**Comment by dacofr** [comment]
In case of https://github.com/fiaas/fiaas-deploy-daemon/pull/262 is ready :smiley:

**Comment by dacofr** [comment]
Closed with #262
